test/integration/replication.test.js

```diff
@@ -41,4 +53,68 @@ describe(Support.getTestDialectTeaser('Replication'), function() {
   it('should be able to make a read', () => {
     return this.User.findAll();
   });
+
+  function setupTransactionTestOptions() {
+    var baseConf = Config[Support.getTestDialect()];
+    var poolEntry = {
+      host: baseConf.host,
+      port: baseConf.port,
+      pool: {}
+    };
+
+    var master = _.clone(poolEntry);
+    master.host = 'master';
+
+    var slave = _.clone(poolEntry);
+    slave.host = 'slave';
+
+    var options = {
+      replication: {
+        write: master,
+        read: [slave]
+      },
+      databaseVersion: '9.1.0' // avoid the db version check (which always hits master)
+    };
+    return options;
+  }
+
+  it('should run read-only transactions on the replica', () => {
+    var sequelize = Support.createSequelizeInstance(setupTransactionTestOptions());
+
+    var resolvedPromise = Utils.Promise.resolve({
+      queryType: 'read'
+    });
+    var connectStub = sandbox.stub(sequelize.connectionManager, '_connect').returns(resolvedPromise);
+
+    return sequelize.transaction({readOnly: true}, (transaction) => {})
+    .catch(() => { // Database doesn't exist, so query *will* fail
```

Comment:
Use `this.sequelize` backed connection and assert with `readOnly: true` read replica gets utilized